docs/DeveloperGuide.md

```diff
@@ -1,34 +1,179 @@
 # Developer Guide
+cover page
+## Table of Contents
+### Target user profile
+Our target audiences are professors who need help organizing their personal work schedule and need more time.
+The professors are pressed for time and they require a simple software to organize their monthly events
+and keep track of their students' attendance and performance. 
+
+### Value proposition
+Our application will reduce the stress of the professor by allowing them to easily enter and store
+their work schedule as well as their students' records. After storing the data, the Professor can have
+quick access to the information in either a list or a calendar view. 
+
+### Audience 
+The design documentation is in general for anyone who wants to understand the system architecture and design of 
+PAC. The following groups are in particular the intended audience of the document.
+- PAC project managers
+- PAC developers
+- PAC software testers
+
+## Architecture
+This section presents the architecture of PAC. It explains the architecture of main components of PAC.
+
+### Overall Architecture
+*Overall Class diagram*
+{To be added in future revisions}
+
+### Command
+![Command](images/Command.png "Class diagram of Command component")         
+*Class diagram of the Command component*  
+
+Commands are the main classes to be executed in PAC. All of the specific Command classes inherit the 
+base Command abstract class, and utilize its abstract execute() method.  
+A subclass of Command is created and executed when the professor input a corresponding command.
+ 
+### Parser
+*Class diagram of the Parser component*  
+There are total of four Parser classes as shown below. Each Parser class correspond to a feature 
```

Comment:
It seems the diagram of parser compont is missing here.If you are going to add it in the feature, I think you can draw a class diagram here and add a sequence diagram to give an example at the end of this part.